src/HamiltonianParameters/HamiltonianParameters.cpp

```diff
@@ -272,5 +272,102 @@ GQCP::TwoElectronOperator HamiltonianParameters::calculateSuperGeneralizedFockMa
+    TwoElectronOperator gc (this->get_g().get_matrix_representation() - lambda*two_op.get_matrix_representation());
+
+    return HamiltonianParameters(this->ao_basis, this->S, hc, gc, this->C);
+}
+
+
+/**
+ *  Constrains and copies the Hamiltonian parameters
+ *
+ *  @param one_op   constraining one electron operator
+ *  @param lambda   lagrangian multiplier for the constraint
+ *
+ *  @return the constrained Hamiltonian parameters
+ */
+HamiltonianParameters HamiltonianParameters::constrain(const GQCP::OneElectronOperator& one_op, double lambda) const {
+
+    OneElectronOperator hc (this->get_h().get_matrix_representation() - lambda*one_op.get_matrix_representation());
+
+    return HamiltonianParameters(this->ao_basis, this->S, hc, this->g, this->C);
+}
+
+
+/**
+ *  Constrains and copies the Hamiltonian parameters
+ *
+ *  @param two_op   constraining two electron operator
+ *  @param lambda   lagrangian multiplier for the constraint
+ *
+ *  @return the constrained Hamiltonian parameters
+ */
+HamiltonianParameters HamiltonianParameters::constrain(const GQCP::TwoElectronOperator& two_op, double lambda) const {
+
+    TwoElectronOperator gc (this->get_g().get_matrix_representation() - lambda*two_op.get_matrix_representation());
+
+    return HamiltonianParameters(this->ao_basis, this->S, this->h, gc, this->C);
+}
+
+
+/**
+ *  Calculates the Mulliken operator for Hamiltonian parameters and a set of GTOs indexes
+ *
+ *  @param gto_list     indexes of the original GTOs on which the Mulliken populations are dependant
+ *
+ *  @return the Mulliken operator for a set of GTOs
+ */
+OneElectronOperator HamiltonianParameters::calculateMullikenOperator(const Vectoru& gto_list) {
+
+
+    if (!this->get_ao_basis()) {
+        throw std::invalid_argument("The Hamiltonian parameters has no underlying GTO basis, Mulliken analysis is not possible.");
+    }
+
+    if (gto_list.size() > this->K) {
+        throw std::invalid_argument("To many GTOs are selected");
+    }
+
+    Eigen::MatrixXd p_a = Eigen::MatrixXd::Zero(this->K, this->K);
+
+    for (size_t index : gto_list) {
+        if (index >= this->K) {
+            throw std::invalid_argument("GTO index is too large");
+        }
+
+        p_a(index, index) = 1;
+    }
+
+    OneElectronOperator S_GTO = this->S;
+    S_GTO.transform(C.inverse());
+
+    Eigen::SelfAdjointEigenSolver<Eigen::MatrixXd> saes_gto (S_GTO.get_matrix_representation());
+    Eigen::SelfAdjointEigenSolver<Eigen::MatrixXd> saes_ov (S.get_matrix_representation());
+
+    Eigen::MatrixXd S_root = saes_ov.operatorSqrt();
+
+    Eigen::MatrixXd S_GTO_root_inverse = saes_gto.operatorInverseSqrt();
+    Eigen::MatrixXd S_GTO_root = saes_gto.operatorSqrt();
+
+
+    Eigen::MatrixXd mulliken_matrix = Eigen::MatrixXd(this->K, this->K);
+
+    mulliken_matrix = (C.adjoint() * S_root * S_GTO_root_inverse * p_a * S_GTO_root * S_root * C + C.adjoint() * S_root * S_GTO_root * p_a * S_GTO_root_inverse * S_root * C)/2 ;
```

Comment:
I know you're just implementing the Mulliken formula, but I think some documentation would be nice. Like: `// Prepare the necessary matrices to calculate the Mulliken matrix` or something before line 353.